browser-extension/mv2/src/background/ExtensionIconManager.js

```diff
@@ -15,9 +15,7 @@ class ExtensionIconManager {
 
   constructor() {
     chrome.tabs.onUpdated.addListener((tabId) => {
-      if (window.tabService.getPageData(tabId, this.#CONSTANTS.PAGE_DATA_ICON_CONFIG)) {
-        this.#updateIconState(tabId);
-      }
+      this.#updateIconState(tabId);
```

Comment:
`updateIconState` checks for page config and uses default state if no config is found in page data. So this check was unnecessary, it prevented reset of state to default state in some cases